mobile/android/android-device.ts

```diff
@@ -185,9 +188,166 @@ export class AndroidDevice implements Mobile.IDevice {
 			this.startPackageOnDevice(packageName).wait();
 			this.$logger.info("Successfully deployed on device with identifier '%s'", this.getIdentifier());
 		}).future<void>()();
-	}
-
-	private ensureFullAccessPermissions(devicePath: string): IFuture<void> {
+    }
+
+    private tcpForward(src: Number, dest: Number): IFuture<void> {
+        return (() => {
+            var tcpForwardCommand = this.composeCommand("forward tcp:%d tcp:%d", src.toString(), dest.toString());
+            this.$childProcess.exec(tcpForwardCommand).wait();
+        }).future<void>()();
+    }
+
+    private startDebuggerClient(port: Number): IFuture<void> {
+        return (() => {
+            var nodeInspectorModuleFilePath = require.resolve("node-inspector");
+            var nodeInspectorModuleDir = path.dirname(nodeInspectorModuleFilePath);
+            var nodeInspectorFullPath = path.join(path.join(nodeInspectorModuleDir, "bin"), "inspector");
+            this.$childProcess.spawn("node", [nodeInspectorFullPath, "--debug-port", port.toString()], { stdio: ["ignore", "ignore", "ignore"], detached: true });
+        }).future<void>()();
+    }
+
+    private openDebuggerClient(url: string): IFuture<void> {
+        return (() => {
+            this.$opener.open(url, "chrome");
```

Comment:
So far only webkit enabled browsers support the debug protocol of node-inspector. On the last meeting we agreed to use Chrome as the most common choice.